tests/test_series.py

```diff
@@ -3352,3 +3353,18 @@ def test_vector_series_normalize():
         [(x, -5, 5), (y, -5, 5), (z, -5, 5)], params={u: 1},
         slice=Plane((0, 0, 0), (0, 1, 0)), normalize=True)
     assert s.normalize
+
+def test_complex_number_eval():
+    xi, wn, x0, v0, t = symbols("xi, omega_n, x0, v0, t")
+    x = Function("x")(t)
+    eq = x.diff(t, 2) + 2 * xi * wn * x.diff(t) + wn**2 * x
+    sol = dsolve(eq, x, ics={x.subs(t, 0): x0, x.diff(t).subs(t, 0): v0})
+    d = {
+        'params' : {
+            wn: 0.5,
+            xi: 0.25,
+            x0: 0.45,
+            v0: 0.0}
+    }
+    s = InteractiveSeries([sol.rhs], [(t, 0, 100)], **d)
+    assert np.all([x != np.nan for x in s.get_data()])
```

Comment:
This is a little bit more tricky. `get_data` returns arrays of numbers. In this particular case it returns two arrays, one containing the data for the x-axis, the other containing data for the y-axis.

Hence, an array is clearly different than `np.nan` and the test succeed, but that's wrong. You need to look that the elements of the array are non NaN.